bindings/azure/blobstorage/blobstorage.go

```diff
@@ -153,14 +156,8 @@ func (a *AzureBlobStorage) Init(metadata bindings.Metadata) error {
 }
 
 func (a *AzureBlobStorage) parseMetadata(metadata bindings.Metadata) (*blobStorageMetadata, error) {
-	connInfo := metadata.Properties
-	b, err := json.Marshal(connInfo)
-	if err != nil {
-		return nil, err
-	}
-
 	var m blobStorageMetadata
-	err = json.Unmarshal(b, &m)
+	err := mapstructure.WeakDecode(metadata.Properties, &m)
```

Comment:
Why is this necessary?

Reply:
Not strictly necessary, but avoids marshalling to JSON and back. It's just more efficient (and the module was already imported in Dapr).